docs/command-line-interface.md

```diff
@@ -0,0 +1,1085 @@
+      --storage-type  (Required) The type of storage to use for the catalog
+      --default-base-location  (Required) Default base location of the catalog
+      --allowed-location  An allowed location for files tracked by the catalog. Multiple locations can be provided by specifying this option more than once.
+      --role-arn  (Required for S3) A role ARN to use when connecting to S3
+      --external-id  (Only for S3) The external ID to use when connecting to S3
+      --tenant-id  (Required for Azure) A tenant ID to use when connecting to Azure Storage
+      --multi-tenant-app-name  (Only for Azure) The app name to use when connecting to Azure Storage
+      --consent-url  (Only for Azure) A consent URL granting permissions for the Azure Storage location
+      --service-account  (Only for GCS) The service account to use when connecting to GCS
+      --remote-url  (For external catalogs) The remote URL to use
+      --property  A key/value pair such as: tag=value. Multiple can be provided by specifying this option more than once
+    Positional arguments:
+      catalog
+```
+
+##### Examples
+
+```
+polaris catalogs create \
+  --storage-type s3 \
+  --default-base-location s3://example-bucket/my_data \
+  --role-arn ${ROLE_ARN} \
+  my_catalog
+
+polaris catalogs create \
+  --storage-type s3 \
+  --default-base-location s3://example-bucket/my_other_data \
+  --allowed-location s3://example-bucket/second_location \
+  --allowed-location s3://other-bucket/third_location \
+  --role-arn ${ROLE_ARN} \
+  my_other_catalog
+```
+
+#### delete
+
+The `delete` subcommand is used to delete a catalog.
+
+```
+input: polaris catalogs delete --help
+options:
+  delete
+    Positional arguments:
+      catalog
+```
+
+##### Examples
+
+```
+polaris catalogs delete some_catalog
+
+polaris catalogs delete another_catalog
+```
+
+#### get
+
+The `get` subcommand is used to retrieve details about a catalog.
+
+```
+input: polaris catalogs get --help
+options:
+  get
+    Positional arguments:
+      catalog
+```
+
+##### Examples
+
+```
+polaris catalogs get some_catalog
+
+polaris catalogs get another_catalog
+```
+
+#### list
+
+The `list` subcommand is used to show details about all catalogs, or those that a certain principal role has access to. The principal used to perform this operation must have the `CATALOG_LIST` privilege.
+
+```
+input: polaris catalogs list --help
+options:
+  list
+    Named arguments:
+      --principal-role  The name of a principal role
+```
+
+##### Examples
+
+```
+polaris catalogs list
+
+polaris catalogs list --principal-role some_user
+```
+
+#### update
+
+The `update` subcommand is used to update a catalog. Currently, this command supports changing the properties of a catalog or updating its storage configuration.
+
+```
+input: polaris catalogs update --help
+options:
+  update
+    Named arguments:
+      --default-base-location  (Required) Default base location of the catalog
```

Comment:
It's said `required`, but the following example doesn't have that. This is not required, right?

`polaris catalogs update --property tag=new_value my_catalog`